agent/agent.go

```diff
@@ -588,79 +578,113 @@ func (r *agent) getStatusFrom(ctx context.Context, member serf.Member, respc cha
 	}
 }
 
-// statusResponse describes a status response from a background process that obtains
-// health status on the specified serf node.
-type statusResponse struct {
-	*pb.NodeStatus
-	member serf.Member
-	err    error
-}
+// timelineUpdateLoop periodically updates the status timeline by comparing
+// changes in local status and merging a timeline from another member.
+func (r *agent) timelineUpdateLoop() {
+	ticker := r.timelineClock.NewTicker(statusUpdateTimeout)
+	defer ticker.Stop()
 
-// recentStatus returns the last known cluster status.
-func (r *agent) recentStatus() (status *pb.SystemStatus, err error) {
-	status, err = r.cache.RecentStatus()
-	if err == nil && status == nil {
-		status = pb.EmptyStatus()
+	for range ticker.Chan() {
```

Comment:
Similar to what I mentioned in the previous PR, I would rewrite this loop to use select statement and accept the context in this method so it can be canceled by the caller.